presto-iceberg/src/main/java/com/facebook/presto/iceberg/HiveTableOperations.java

```diff
@@ -450,4 +455,39 @@ private static int parseVersion(String metadataLocation)
             return -1;
         }
     }
+
+    /**
+     * Deletes the oldest metadata files if {@link
+     * TableProperties#METADATA_DELETE_AFTER_COMMIT_ENABLED} is true.
+     *
+     * @param base table metadata on which previous versions were based
+     * @param metadata new table metadata with updated previous versions
+     */
+    private void deleteRemovedMetadataFiles(TableMetadata base, TableMetadata metadata)
+    {
+        if (base == null) {
+            return;
+        }
+
+        boolean deleteAfterCommit =
+                metadata.propertyAsBoolean(
+                        TableProperties.METADATA_DELETE_AFTER_COMMIT_ENABLED,
+                        TableProperties.METADATA_DELETE_AFTER_COMMIT_ENABLED_DEFAULT);
+
+        if (deleteAfterCommit) {
+            Set<MetadataLogEntry> removedPreviousMetadataFiles =
+                    Sets.newHashSet(base.previousFiles());
+            // TableMetadata#addPreviousFile builds up the metadata log and uses
+            // TableProperties.METADATA_PREVIOUS_VERSIONS_MAX to determine how many files should stay in
+            // the log, thus we don't include metadata.previousFiles() for deletion - everything else can
+            // be removed
```

Comment:
This comment was confusing to me at first because `TableMetadata#addPreviousFile` isn't referenced in this logic, but rather indirectly when generating `previousFiles`. I think we can rewrite it to be a bit more clear

```suggestion
            // metadata.previousFiles() is populated using  TableProperties.METADATA_PREVIOUS_VERSIONS_MAX
            // thus we don't include  metadata.previousFiles() for deletion - everything else can be removed
```

Reply:
Thanks for the suggestion, it makes the meaning of the comment more clear. Fixed, please take a look when available!